homework/index.js

```diff
@@ -30,13 +30,75 @@
     return elem;
   }
 
+  function pageConstructor(data) {
+    data.sort((a, b) => a.name.localeCompare(b.name));
+    const root = document.getElementById('root');
+    const headerDiv = createAndAppend('div', root, { class: 'headerDiv' });
+    createAndAppend('p', headerDiv, { text: 'HYF Repositories' });
+    const headerSelect = createAndAppend('select', headerDiv, { class: 'headerSelect' });
+    data.map((elem, index) =>
+      createAndAppend('option', headerSelect, { text: elem.name, value: index }),
+    );
+    const container = createAndAppend('div', root, { class: 'container' });
+    const repoInfoDiv = createAndAppend('div', container, { class: 'repoInfoDiv' });
+    const ul = createAndAppend('ul', repoInfoDiv);
+    const repository = createAndAppend('li', ul, { text: `Repository :` });
+    createAndAppend('a', repository, {
+      text: `${data[0].name}`,
+      href: data[0].html_url,
+      target: '_blank',
+      id: 'repository',
+    });
+    createAndAppend('li', ul, { text: `Description :${data[0].description}`, id: 'description' });
+    createAndAppend('li', ul, { text: `Forks :${data[0].forks}`, id: 'forks' });
+    createAndAppend('li', ul, { text: `Update :${data[0].updated_at}`, id: 'update' });
+    const contributorDiv = createAndAppend('div', container, { class: 'contributorDiv' });
+    createAndAppend('p', contributorDiv, { text: 'Contributions' });
+    const contributorUl = createAndAppend('ul', contributorDiv, { class: 'contributorUl' });
+    function contributor() {
+      fetchJSON(data[headerSelect.value].contributors_url, (err, dt) => {
+        if (err) {
+          createAndAppend('div', root, { text: err.message, class: 'alert-error' });
```

Comment:
At the end of your if statement, if you got an error it will give another error!!

**Why is that?**
because the javascript interpreter won't stop on just creating a new element on the page here in line 61, but instead javascript will continue reading until it finishes this `index.js` document.

and the other error speaking about, is because there is no `dt` content but it will be `undefined` _(no response received)_ and that means it will have no content to loop through

**solution?** just extend your `else` close to the end of the function to contain every appending and it will prevent the error 😃